### PR TITLE
chat window and nav bar

### PR DESCRIPTION
**Pull request by boybutcher**
conditional rendering of chat window, general chat window performance update

Intend on moving login/signup/username in nav bar to be inline with logo and make search bar float right

### CODE REVIEW

**Comment by boybutcher** [comment]
I also am thinking I should make both a minimize and close feature for the chat window, as far as getting the receivers name, i havent really thought it through to be completely honest

File: client/src/components/WriteMessage.jsx [hunk 1]
```diff
@@ -55,42 +53,22 @@ class WriteMessage extends React.Component {
...
-  //   .catch((error) => {
-  //     console.log('retrieveMessages failed! error: ', error);
-  //   })
-  // }
-
-  // componentDidMount() {
-  //   setTimeout(this.retrieveMessages, 1500);
-  // }  
+  scrollToBottom() {
+    const node = ReactDOM.findDOMNode(this.messagesEnd);
```
Comment (abianco3) [inline client/src/components/WriteMessage.jsx]: I've never used this method; it looks rad. Does it give you a reference to the actual DOM node for any React component?

Comment (abianco3) [inline client/src/components/WriteMessage.jsx]: I vote for minimization ability for sure.

Comment (boybutcher) [inline client/src/components/WriteMessage.jsx]: i believe it does, I put a div that will always be at the bottom of the chat log so when the component updates, itll jump straight to it in the chat window

Comment (boybutcher) [inline client/src/components/WriteMessage.jsx]: let me send you the stack overflow page if i can find it

Comment (boybutcher) [inline client/src/components/WriteMessage.jsx]: http://stackoverflow.com/questions/37620694/how-to-scroll-to-bottom-in-react

it was pretty dope

Comment (abianco3) [inline client/src/components/WriteMessage.jsx]: thanks Bryan

**Comment by boybutcher** [comment]
CLOSING THIS, CHECKOUT OTHER PR

**Pull request closed without merging**